### ISSUE TITLE
expose optimizations as browserify transform, plugin

### ISSUE THREAD

**Issue by yoshuawuyts** [bendrucker/bundleify#5]
I reckon it'd be cool if the transforms in bundleify could be exposed as a single transform. Perhaps this needs to be a separate package tho, but I quite like the idea of doing:

``` sh
$ browserify -g bundleify/transform -p bundleify/plugin | bundleify/compress
```

Or something along those lines. Makes it easier to point people to a single repo as "the solution to browserify optimizations"; with app-specific transforms as another thing (like `brfs` and `yo-yoify`). 

The downside of this being that `bundleify` ships with `browserify` included, which kind of goes against this, but I feel having common ground for optimizating builds would be grand. Thoughts?

**Comment by yoshuawuyts** [comment]
Oh, this could perhaps also tie into #4, as the CLI would then just be `browserify`

**Comment by bendrucker** [comment]
The transform bit could get a bit weird. You'd basically have to export a function of filename/options and then return your own transform stream that pipelines all the transforms.

And then the problem with compression is that I'm using uglifyify as a transform because Uglify's own source map support is too buggy: 834e6d788d53d6d33c18d31f77ad5d9c15abec04. 

I'd love to logically split the lib up but right now the originally proposed API would be impossible.

**Comment by yoshuawuyts** [comment]
ah, so to get it straight - you're no longer doing a second pass of compression using `uglify-js`? Think the other issues are surmountable, though probably might be out of scope for this package

**Comment by bendrucker** [comment]
> ah, so to get it straight - you're no longer doing a second pass of compression using uglify-js

Correct. I'd like to bring it back but source maps and uglify don't play well together at the moment.

**Comment by yoshuawuyts** [comment]
Cool, thanks for clarifying!

**Comment by ahdinosaur** [comment]
ooh, splitting out any transforms / plugins here to be re-usable elsewhere is a very interesting idea.

**Comment by bendrucker** [comment]
I can definitely take a stab at splitting out the transform and plugin steps to be individually consumable
